packages/pwa/app/utils/url.js

```diff
@@ -185,3 +191,43 @@ export const removeQueryParamsFromPath = (path, keys) => {
 
     return `${pathname}${paramStr && '?'}${paramStr}`
 }
+
+/**
+ * Rebuild the path by adding/removing locale/site query params based on url config
+ * @param {string} url - based url of the output url
+ * @param {object} configValues - object that contains values of url param config
+ * @return {string} - an output url
+ *
+ * @example
+ * //pwa-kit-config.json
+ * url {
+ *    locale: "query_param"
+ * }
+ * buildPathWithLocaleAndSiteParams('/women/dresses/', {locale: 'en-GB'})
+ *
+ * Returns
+ *  /women/dresses/?locale=en-GB
+ *
+ */
+export const buildPathWithUrlConfigParams = (url, configValues = {}) => {
+    const urlConfig = getUrlConfig()
+    const {locale: localeParamType, siteId: siteIdParamType} = urlConfig
+    const {locale, siteId} = configValues
+    const queryParams = {}
+
+    if (localeParamType === urlParamTypes.QUERY_PARAM) {
```

Comment:
is there another option for `localeParamType` besides `query_param`?

Reply:
Currently, we'd provide three options: `path`, `query_param`, `none`. 

- path: this means the param to be in the path
- query_param: this means the param will be in the query parameter
- none: don't show the param in the url at all.

This util only wants to create the path to at the url values as query param since path has been taken care by `basename` being passed to Router if they choose to config that way